xkcd/pkg/web/web.go

```diff
@@ -0,0 +1,60 @@
+package web
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/vespian/go-exercises/xkcd/pkg/types"
+)
+
+func fetchStory(url string) (types.Story, error) {
```

Comment:
if you return a pointer to types.Story you can drop all the annoying `return types.Story{}, ...` bits.